lib/transaction.js

```diff
@@ -32,6 +32,11 @@ class Transaction extends EventEmitter {
         this.initRejectFn = reject;
         container = (transactor) => {
           resolve(transactor);
+          // because user must await executionPromise in same tick
+          // todo^ is this true?
+          transactor.executionPromise = transactor.executionPromise.catch((err) => {
+            throw err;
+          });
```

Comment:
Hmm... I suspect that this isn't actually addressing the underlying race condition.  Instead, it is just adding a slight delay in the event processing queue, which makes the race condition less likely to occur.

Reply:
Yes, this change is wrong too